src/cameraGeom/Camera.cc

```diff
@@ -0,0 +1,138 @@
+        auto det = camera[fromSys.getDetectorName()];
+        return det->getTransformMap()->getTransform(fromSys, toSys);
+    } else if (toSys.hasDetectorName()) {
+        auto det = camera[toSys.getDetectorName()];
+        return det->getTransformMap()->getTransform(fromSys, toSys);
+    } else {
+        return camera.getTransformMap()->getTransform(fromSys, toSys);
+    }
+}
+
+} // anonymous
+
+Camera::Camera(std::string const &name, DetectorList const &detectorList,
+               std::shared_ptr<TransformMap> transformMap, std::string const &pupilFactoryName) :
+    DetectorCollection(detectorList),
+    _name(name), _transformMap(std::move(transformMap)), _pupilFactoryName(pupilFactoryName)
+    {}
+
+Camera::~Camera() noexcept = default;
+
+Camera::DetectorList Camera::findDetectors(lsst::geom::Point2D const &point,
+                                           CameraSys const &cameraSys) const {
+    auto transform = getTransformFromOneTransformMap(*this, cameraSys, getNativeCameraSys());
+    auto nativePoint = transform->applyForward(point);
+
+    DetectorList detectorList;
+    for (auto const &item : getIdMap()) {
+        auto detector = item.second;
+        auto nativeToPixels = detector->getTransform(getNativeCameraSys(), PIXELS);
+        auto pointPixels = nativeToPixels->applyForward(nativePoint);
+        if (lsst::geom::Box2D(detector->getBBox()).contains(pointPixels)) {
+            detectorList.push_back(std::move(detector));
+        }
+    }
+    return detectorList;
+}
+
+std::vector<Camera::DetectorList> Camera::findDetectorsList(std::vector<lsst::geom::Point2D> const &pointList,
+                                                            CameraSys const &cameraSys) const {
+    auto transform = getTransformFromOneTransformMap(*this, cameraSys, getNativeCameraSys());
+    std::vector<DetectorList> detectorListList(pointList.size());
+
+    auto nativePointList = transform->applyForward(pointList);
+
+    for (auto const &item: getIdMap()) {
+        auto const &detector = item.second;
+        auto nativeToPixels = detector->getTransform(getNativeCameraSys(), PIXELS);
+        auto pointPixelsList = nativeToPixels->applyForward(nativePointList);
+        for (std::size_t i = 0; i < pointPixelsList.size(); ++i) {
+            auto const &pointPixels = pointPixelsList[i];
+            if (lsst::geom::Box2D(detector->getBBox()).contains(pointPixels)) {
+                detectorListList[i].push_back(detector);
+            }
+        }
+    }
+    return detectorListList;
+}
+
+std::shared_ptr<afw::geom::TransformPoint2ToPoint2> Camera::getTransform(CameraSys const &fromSys,
+                                                                         CameraSys const &toSys) const {
+    try {
+        return getTransformMap()->getTransform(fromSys, toSys);
+    } catch (pex::exceptions::InvalidParameterError &) {}
+    // Must be an old-style Camera that doesn't share its TransformMap with its
```

Comment:
"Old" here refers to DM-14980, which changed the `makeCamera*` methods in give the `Camera` and all of its `Detectors` a common `TransformMap`.  in that case, the above initial lookup should always succeed because this `TransformMap` has all of the transforms (unless the arguments are totally bogus, and then we'll throw a exception again below).

But if you construct a `Camera` more manually, it's possible to create one in which the `Camera` itself has a `TransformMap` with no `Detector`-specific transforms and each `Detector` has a `TransformMap` with only transforms relevant to that specific `Detector`, and then we need the code below.

I'll paste all of that into a more extensive code comment.